cairo-libs/packages/ics23/src/types.cairo

```diff
@@ -47,17 +49,18 @@ impl ProofAsProtoMessage of ProtoMessage<Proof> {
         }
     }
 
-    fn decode_raw(ref self: Proof, ref context: DecodeContext) {
-        match self.clone() {
-            Proof::Exist(mut p) => {
-                p.decode_raw(ref context);
-                self = Proof::Exist(p);
-            },
-            Proof::NonExist(mut p) => {
-                p.decode_raw(ref context);
-                self = Proof::NonExist(p);
-            },
+    fn decode_raw(ref context: DecodeContext) -> Option<Proof> {
+        let exist: Option<ExistenceProof> = context.decode_raw();
+        if exist.is_some() {
+            return Option::Some(Proof::Exist(exist.unwrap()));
```

Comment:
don't we have pattern matching in Cairo. Avoid calling `unwrap()` unnecessarily. Use `match exist { Some(exist) => ... }` to return here without `unwrap()`.

Too bad, we can't have `if let Some(exist) = exist { ... }`.

Reply:
I think, this implementation is wrong. When it's `oneof`, it's still encoded with field number, corresponding to the oneof field number.

Did you test this?

Reply:
I see you merged the branch. Tackle this in #282